packages/app/src/cli/services/function/binaries.ts

```diff
@@ -22,7 +22,8 @@ class DownloadableBinary {
   constructor(name: string, version: string, gitHubRepo: string) {
     this.name = name
     this.version = version
-    this.path = joinPath(dirname(fileURLToPath(import.meta.url)), '..', 'bin', `${this.name}`)
+    const filename = process.platform === 'win32' ? `${name}.exe` : name
```

Comment:
I would suggest we leave open the possibility of fixing this in the release itself, so the filename itself gets the `.exe` suffix. So I'd suggest something like this:

```suggestion
    const needsExe = process.platform === 'win32' && !name.endsWith('.exe')
    const filename = needsExe ? `${name}.exe` : name
```

This would prevent a situation like `filename.exe.exe`

Reply:
Sorry I'm confused. _name_ is _always_ going to be set to one of `javy` or `function-runner` (there are only two callsites to the constructor, one [here](https://github.com/Shopify/cli/blob/09386a083ce8773e320d0dfb4522c80130a6b42b/packages/app/src/cli/services/function/binaries.ts#L82) and one [here](https://github.com/Shopify/cli/blob/09386a083ce8773e320d0dfb4522c80130a6b42b/packages/app/src/cli/services/function/binaries.ts#L89) and the _name_ argument is hardcoded in both) so `!name.endsWith('.exe')` will always evaluate to `true` which would simplify `needsExe` to `process.platform === 'win32'`.

Reply:
My point is that we should consider changing the file in the release to be a gzipped `.exe` file instead of a gzipped binary with no extension. But looking at the broader code, we do seem to decide the output filename not based on the release filename but based on the local hardcoded value. So this can certainly stay as is.